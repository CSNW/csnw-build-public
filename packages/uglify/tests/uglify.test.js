const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

const uglify = require('../')();
const cwd = join(__dirname, 'fixtures');

test('skip pipeline in development', async () => {
  const process = await uglify.pipeline({
    mode: 'development',
    cwd,
    sourcemaps: true,
    cache: false
  });
  const files = await Files.from('*', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});

test('pipeline', async () => {
  const process = await uglify.pipeline({
    mode: 'production',
    cwd,
    sourcemaps: true,
    cache: false
  });
  const files = await Files.from('*', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
