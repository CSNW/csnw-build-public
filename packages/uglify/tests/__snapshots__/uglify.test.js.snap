// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "a.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/uglify/tests/fixtures",
    "contents": "function test(){return\\"done.\\"}",
    "cwd": "C:/dev/csnw-build/packages/uglify/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/uglify/tests/fixtures/a.js",
    "sourceMap": Object {
      "file": "a.js",
      "mappings": "AAAA,SAASA,OAQP,MADe",
      "names": Array [
        "test",
      ],
      "sources": Array [
        "a.js",
      ],
      "sourcesContent": Array [
        "function test() {
  const name = \`a\`;

  if (name !== 'a') {
    console.log('not a');
  }

  const result = \`done.\`;
  return result;
}
",
      ],
      "version": 3,
    },
  },
  "b.html": Object {
    "baseDir": "C:/dev/csnw-build/packages/uglify/tests/fixtures",
    "contents": "<title>B</title>",
    "cwd": "C:/dev/csnw-build/packages/uglify/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/uglify/tests/fixtures/b.html",
    "sourceMap": null,
  },
}
`;

exports[`skip pipeline in development 1`] = `
Object {
  "a.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/uglify/tests/fixtures",
    "contents": "function test() {
  const name = \`a\`;

  if (name !== 'a') {
    console.log('not a');
  }

  const result = \`done.\`;
  return result;
}
",
    "cwd": "C:/dev/csnw-build/packages/uglify/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/uglify/tests/fixtures/a.js",
    "sourceMap": null,
  },
  "b.html": Object {
    "baseDir": "C:/dev/csnw-build/packages/uglify/tests/fixtures",
    "contents": "<title>B</title>",
    "cwd": "C:/dev/csnw-build/packages/uglify/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/uglify/tests/fixtures/b.html",
    "sourceMap": null,
  },
}
`;
